src/jekyll/spark-submit.md

```diff
@@ -0,0 +1,599 @@
+
+If we really want to apply some configuration say e.g. on the Executors, one option would to hook in e.g. the `KubernetesClusterManager` or still better to use the `executorInitContainerBootstrap` to achieve the needed job (if any...).
+
+*[TBC] This is the main point to be discussed... - As first instance, we can don't need the Driver configuration steps*
+
+*[TBC]: Note on performance: At first sight, it seems the `client-mode` is a bit slower than the `cluster-mode` for actions only run on driver side - To be benchmarked...*
+
+# Hadoop
+
+We want to access HDFS from Driver and Executor in whatever state we are.
+
+In `cluster-mode`, this is achived mounting an existing hadoop `configmap` in the Driver. A new confimap is created based on the given one with that code in the `DriverConfigurationStepsOrchestrator`. This is in a way needed as the driver is not created via a end-user controlled process, so the end-user can not enforce its own configmaps on the Driver pod. Spark must in a way propagate the configmaps.
+
+```scala
+val hadoopConfigSteps =
+  hadoopConfDir.map { conf =>
+    val hadoopStepsOrchestrator =
+      new HadoopStepsOrchestrator(
+        kubernetesResourceNamePrefix,
+        namespace,
+        hadoopConfigMapName,
+        submissionSparkConf,
+        conf)
+```
+
+However, in `cluster-mode`, we simply have to ensure that the Pod where we start the Spark process is created with a correct Hadoop configmap. "By magic", the Executors benefit from that configuration (you can check this with  `spark.sparkContext.hadoopConfiguration`) and Spark can access HDFS in a distributed way.
+
+```scala
+spark.sparkContext.hadoopConfiguration.get("fs.defaultFS")
+res19: String = hdfs://hdfs-k8s-hdfs-k8s-hdfs-nn:9000/
+```
+
+Please not the slight difference if you isntanciate your self the Hadoop Config (this may be a limitation of the `client-mode` if you want to directly address the Hadoop API from within an Executor).
+
+```scala
+// Via cluster mode, we have the full config as the hadoop conf dir is mounted by configmap propagation.
+(1 to 3).toDS.coalesce(3).map(_ => { new org.apache.hadoop.conf.Configuration().get("fs.defaultFS")}).collect
+res0: Array[String] = Array(hdfs://hdfs-k8s-hdfs-k8s-hdfs-nn:9000/, hdfs://hdfs-k8s-hdfs-k8s-hdfs-nn:9000/, hdfs://hdfs-k8s-hdfs-k8s-hdfs-nn:9000/)
+```
+
+```scala
+// Via cluster mode, we do not have full config as the hadoop conf dir is not mounted by configmap propagation but only available in the Spark Context.
+(1 to 3).toDS.coalesce(3).map(_ => { new org.apache.hadoop.conf.Configuration().get("fs.defaultFS")}).collect
+res16: Array[String] = Array(file:///, file:///, file:///)
+```
+
+# Dependencies
+
+For `cluster-mode`, we have the configuration orchestrator and its Steps which ensure that when `--jars` or `-Dspark.jars` are defined (same reaseonaring applies for `--files`):
```

Comment:
this comment really goes from all cluster managers with spark, just came to my mind as I was reading this and something you might want to consider.

one thing which has tripped up a ton of users is understanding *when* jars become available and how you override various things.  As an extreme example, you can't use "--jars" to replace the code of spark's Executor, as the executor itself has already started by that point.  (Nobody would want to do that, but there are plenty of cases where you might want to replace some dependency, and its tricky because sometimes they are loaded before they you get to downloading and loading "--jars".)